django-s3-file-field-client/src/client.ts

```diff
@@ -19,16 +19,38 @@ interface UploadedPart {
   size: number;
   etag: string;
 }
+
+export enum UploadResultState {
+  Aborted,
+  Successful,
+  Error,
+}
 // Return value from uploadFile()
 export interface UploadResult {
   value: string;
-  state: 'aborted' | 'successful' | 'error';
+  state: UploadResultState;
 }
 
-export default class S3FFClient {
-  protected readonly baseUrl: string;
+export enum ProgressState {
+  Initializing,
+  Sending,
+  Finalizing,
+  Done,
+}
 
-  constructor(baseUrl: string) {
+export interface ProgressEvent {
+  readonly uploaded?: number;
+  readonly total?: number;
+  readonly state: ProgressState;
+}
+
+type ProgressCallback = (progress: ProgressEvent) => void;
+
+export default class S3FFClient {
+  constructor(
+    protected readonly baseUrl: string,
+    private readonly onProgress: ProgressCallback = () => { /* no-op*/ },
```

Comment:
@zachmullen This should be `protected`.

I'll push a fix as part of another PR.